plugin/src/templates/edge/utils.ts

```diff
@@ -34,6 +36,21 @@ export const addMiddlewareHeaders = async (
   })
   return response
 }
+
```

Comment:
These are re-declared here, because the other declaration is Node code which depends on the `next` package. We just need the types here, and it's fine for our purposes to pretend that `NextResponse`s are actually just `Response`s.

Reply:
[dust] Could we leave a comment about why it's fine for us to pretend that `NextResponse`s are just `Response`s in the code itself? 

I worry that there might be some confusion at a later point in time over why `Response` isn't `NextResponse`. Based on what you're saying I'm assuming it's because we don't need access to the other properties and methods that `NextResponse` has available at this time.